### PR TITLE
Introduce Teams support

### PR DESCRIPTION
**Pull request by tg123**
image

### CODE REVIEW

File: notify/teams/teams.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	return nil
+}
+
+// SendTeamsMessage sends the message to the teams channel
+func (c *NotifyConfig) SendTeamsMessage(title, msg string) error {
+	msgCard := goteamsnotify.NewMessageCard()
+	msgCard.Title = title
+	msgCard.Text = msg
+
+	return c.client.Send(c.WebhookURL, msgCard)
```
Comment (haoel) [inline notify/teams/teams.go:61]: Should we deal with time out problem?

File: README.md [hunk 2]
```diff
@@ -188,6 +189,9 @@ notify:
       key: xxxxxxxxxxxx # yunpian apikey
       mobile: 123456789,987654321 # mobile phone number, multiple phone number joint by `,`
       sign: "xxxxxxxx" # need to register; usually brand name
+  teams:
+      - name: "teams alert service"
+        webhook: "https://outlook.office365.com/webhook/a1269812-6d10-44b1-abc5-b84f93580ba0@9e7b80c7-d1eb-4b52-8582-76f921e416d9/IncomingWebhook/3fdd6767bae44ac58e5995547d66a4e4/f332c8d9-3397-4ac5-957b-b8e3fc465a8c" # see https://docs.microsoft.com/en-us/outlook/actionable-messages/send-via-connectors
```
[the remark this answers is not in the document]
Comment (tg123) [reply README.md:194]: hide in yaml?

File: notify/teams/teams.go (hunk 1 above)
Comment (tg123) [reply notify/teams/teams.go:61]: I wonder why there is no `ctx`, I was opening an issue but too many fields in the future req template :(

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (haoel) [reply README.md:194]: No, I mean the webhook url probably expose your private information, it's better to hide the privacy. Unless you are sure it is not a problem.

[the remark this answers is not in the document]
Comment (tg123) [reply README.md:194]: this is the url in public doc, no worries

**Review by haoel — APPROVED** [review]
LGTM

**Pull request merged**